src/jira-connector.ts

```diff
@@ -22,26 +22,33 @@ export class JiraConnector {
     });
   }
 
-  async getTicketDetails(key: string): Promise<JIRADetails> {
-    const issue: JIRA.Issue = await this.getIssue(key);
-    const {
-      fields: { issuetype: type, project, summary },
-    } = issue;
-
-    return {
-      key,
-      summary,
-      url: `${this.JIRA_BASE_URL}/browse/${key}`,
-      type: {
-        name: type.name,
-        icon: type.iconUrl,
-      },
-      project: {
-        name: project.name,
-        url: `${this.JIRA_BASE_URL}/browse/${project.key}`,
-        key: project.key,
-      },
-    };
+  async getTicketDetails(key: string): Promise<any> {
```

Comment:
@cakeinpanic I had to change the return of this function to `any`. If there is an error and the control goes to the catch block, we get error since now we are not returning any data back of type `JIRADetails`. It would be `undefined`. So I am not sure if there is any elegant way to do this. If yes let me know. Another option would be `Promise<JIRADetails | undefined>`

Reply:
you should keep `JIRADetails`, no need to specify output in case of method throwing an error
look at the example
```
function a():string {
throw new Error()
}
const b = a()
```

in this case `b` would not be assigned to undefined

Reply:
now I see that you had a typescript error
check the fix
https://github.com/cakeinpanic/jira-description-action/commit/4fdeade0ca9beb9c2b29d6795dab75bb12573d83

Reply:
Got it @cakeinpanic. So basically we can move the throw out of catch. Will try it out in my local.